lib/request/complement.js

```diff
@@ -79,96 +77,135 @@ function findPartialIntersections(requestedPath, optimizedPath, requestTree) {
     }
 
     if (depthDiff === 0) {
-        return recurse(requestTree, 0, [], []);
+        return recurse(requestedPath, optimizedPath, requestTree, 0, [], []);
     } else if (depthDiff > 0) {
-        return recurse(requestTree, 0, arraySlice(requestedPath, 0, depthDiff), []);
+        return recurse(requestedPath, optimizedPath, requestTree, 0, requestedPath.slice(0, depthDiff), []);
     } else {
-        return recurse(requestTree, -depthDiff, [], arraySlice(optimizedPath, 0, -depthDiff));
+        return recurse(requestedPath, optimizedPath, requestTree, -depthDiff, [], optimizedPath.slice(0, -depthDiff));
     }
+}
 
-    function recurse(currentTree, depth, rCurrentPath, oCurrentPath) {
```

Comment:
The change here is that `recurse` becomes a function declaration instead of a closure. It seems to only close over two variables which are now passed in directly. This saves a function object per call of `findPartialIntersections` + an activation per call of `recurse`.